frontend/src/pages/Main/Main.tsx

```diff
@@ -59,13 +60,26 @@ export const Main = (props: MainProps) => {
     isPlaying ? Math.ceil(1000 / speed) : null,
   )
 
+  const setDataIdx = (idx: number) => {
+    const newData: dataVal[] = []
+    for (let i = 0; i < idx; i++) {
+      const newInstructions = props.instructions[i].variable_changes
+      for (const [key, value] of Object.entries(newInstructions)) {
+        updateData(key, value, newData)
+      }
+    }
+    setData(newData)
+    setCurIdx(idx)
+  }
```

Comment:
This might need to be revisited in the future, if we want to support large number of instructions (currently not a priority), as it might be too slow.

Reply:
Ok will raise an issue for it